### PR TITLE
Ad nlohmann-json as dependency of the superbuild

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/robotology-superbuild/issues/763 . I added it as a dependency in all points, I am just missing https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/pull/45 to also bump the version of vcpkg dependencies used in CI.

### CODE REVIEW

**Comment by traversaro** [comment]
> I am just missing [robotology/robotology-superbuild-dependencies-vcpkg#45](https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/pull/45) to also bump the version of vcpkg dependencies used in CI.

This was fixed by https://github.com/robotology/robotology-superbuild/pull/776/commits/de82b26cf2f3c478f157a1f2348d36006b961a95 . As the vcpkg tarball has shown to be more stable lately, I switched CI to also use latest as originally suggested by @S-Dafarra, for consistency with the user docs.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -406,6 +413,11 @@ jobs:
         cd build
         cmake -DROBOTOLOGY_ENABLE_DYNAMICS_FULL_DEPS:BOOL=OFF .
 
+    - name: Install dependencies unsupported in Ubuntu 18.04
+      if: contains(matrix.os, '20.04')
```
Comment (S-Dafarra) [inline .github/workflows/ci.yml:417]: Why the condition is different here?

Comment (traversaro) [reply .github/workflows/ci.yml:417]: Because it is for installing dependencies that are not available on 18.04, so it is run on Ubuntu 20.04 .

Comment (S-Dafarra) [reply .github/workflows/ci.yml:417]: Sorry, I should have been more precise. I meant with respect to the similar if clause above at  line 231

Comment (S-Dafarra) [reply .github/workflows/ci.yml:417]: I saw later that one applies to docker images and this to os. Not important at all

**Review by S-Dafarra — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (traversaro) [reply .github/workflows/ci.yml:417]: > 
> 
> I saw later that one applies to docker images and this to os. Not important at all

Yes, that's it. The clause are different as for the docker images you need to check the used docker image, not the base os.

**Comment by traversaro** [comment]
The Conda/Linux CI failure are unrelated (see https://github.com/robotology/robotology-superbuild/issues/785), I think once the other jobs finish we can merge.

**Pull request merged**